tensorrt_llm/_torch/pyexecutor/guided_decoder.py

```
@@ -71,109 +175,83 @@ def __init__(self,
                 matcher = self.grammar_matcher_factory.create(
-                    llm_req.guided_decoding_params)
+                    req.guided_decoding_params)
                 self.grammar_matchers[slot] = matcher
 
             if matcher_advance:
                 matcher = self.grammar_matchers[slot]
-                # The last new token must be acceptable unless the matcher is terminated in a drafting loop.
-                if llm_req.py_is_draft and (matcher.is_terminated()
-                                            or self.is_draft_terminated[slot]):
+                # The last new token must be acceptable unless the matcher is terminated:
+                # 1. For the main model loop, when overlap scheduler is enabled, the matcher may have accepted the EOS token in the draft tokens at the previous iteration.
```

Ruff (0.12.2)

Line too long (171 > 120)

(E501)